filters/builtin/creationmetrics.go

```diff
@@ -31,10 +31,14 @@ func NewRouteCreationMetrics(metrics filters.Metrics) *RouteCreationMetrics {
 
 // Do implements routing.PostProcessor and records the filter creation time.
 func (m *RouteCreationMetrics) Do(routes []*routing.Route) []*routing.Route {
+	return m.reportRouteCreationTimes(routes, time.Now())
+}
+
+func (m *RouteCreationMetrics) reportRouteCreationTimes(routes []*routing.Route, now time.Time) []*routing.Route {
 	for _, r := range routes {
 		for origin, start := range m.startTimes(r) {
 			if m.initialized {
-				m.metrics.MeasureSince(metricsPrefix+origin, start)
+				m.metrics.MeasureFloat(metricsPrefix+origin, now.Sub(start).Seconds()/100)
```

Comment:
Can't you shift the duration instead of converting to float?
See https://play.golang.org/p/eXWe0zFXYge

Reply:
done